src/Microsoft.AspNet.Diagnostics/Views/BaseView.cs

```diff
@@ -117,6 +118,48 @@ protected void WriteLiteral(object value)
                 values);
         }
 
+        private List<string> _attributeValues { get; set; }
+
+        protected void WriteAttributeValue(string thingy, int startPostion, object value, int endValue, int dealyo, bool yesno)
+        {
+            if(_attributeValues == null)
+            {
+                _attributeValues = new List<string>();
+            }
+
+            _attributeValues.Add(value.ToString());
+        }
+
+        private string _attributeEnding { get; set; }
+
+        protected void BeginWriteAttribute(string name, string begining, int startPosition, string ending, int endPosition, int thingy)
+        {
+            if(!string.IsNullOrEmpty(_attributeEnding))
+            {
+                throw new InvalidOperationException();
+            }
+
+            Output.Write(begining);
+            _attributeEnding = ending;
+        }
+
+        protected void EndWriteAttribute()
+        {
+            if(string.IsNullOrEmpty(_attributeEnding))
+            {
+                throw new InvalidOperationException();
```

Comment:
It's odd to throw an exception with no message. So what I'm asking is whether this is effectively an assertion (as in, "this can't possibly happen"), or if there is a reasonable way where this *could* happen, in which case it should have a message.

If it's really more of an assertion then this should just be an assert, yes.